results/29-viper-drug-target-enrichment.Rmd

```diff
@@ -0,0 +1,173 @@
+---
+title: "Tumor type association with latent variable correlated viper protein expression in NF tumors"
+author: "Robert Allaway"
+date: "`r Sys.Date()`"
+output:
+  html_document:
+    code_folding: hide
+    toc: true
+    toc_float: true
+---
+
+```{r global_options, include=FALSE}
+knitr::opts_chunk$set(warning=FALSE, message=FALSE)
+```
+
+
+Here, we are evaluating the 'druggability' of viper proteins that are well correlated with variables. 
+The plots at the end of this are labeled by LVs, but the enrichment is actually based on the list of viper proteins that have a correlation of 0.5 or greater with LV expression across all tumors.
+
+# Import packages
+
+First, import packages to process and plot the data. 
+
+```{r message=FALSE, warning=FALSE}
+library(dplyr)
+library(tidyr)
+library(purrr)
+library(synapser)
+synLogin()
+library(clusterProfiler)
+
+viper_res <- synTableQuery('SELECT * FROM syn21260871 where numSamps = 77')$filepath %>% readr::read_csv()
+
+drug_targets <- feather::read_feather(synGet('syn20700199')$path)
+
+drug_targets <- drug_targets %>% 
+  filter(mean_pchembl > 7) %>% 
+  mutate(gene= hugo_gene) %>% 
+  select(gene, std_name) 
+```
+
+# Significance testing
+
+First, create a list of all druggable genes. We're using the drug-target explorer data, for any gene for which there is drug-target relationship with a mean_pchembl value >7, which corresponds to 1 uM. This gives us a rough approximation of druggable targets in the human genome but could be a bit under conservative.
+
+Then, we'll select the viper proteins that are 0.65 or greater correlation with latent variable expression, by latent variable (so each group of genes is based on a single latent variable expression, where each gene in the group is a highly-correlated viper protein)- this is basd on the threshold in notebook 16. 
+
+Then, we'll perform a weighted Kolmogorov-Smirnov test using the `GSEA` function from the `clusterProfiler` package to assess whether any of the viper genes are enriched in the universe of druggable genes. Here, we're treating each LV as the ranked list of viper genes (for standard GSEA, this would be the differentially expressed genes, for example) and treating the list of all drug targets as the gene set we're looking for enrichment in. 
+
+
+```{r message=FALSE, warning=FALSE}
+
+term2gene <- drug_targets %>% 
+  mutate(term = 'all_drugs') %>% 
+  select(term, gene) %>% 
+  distinct()
+
+tidy <- viper_res %>% 
+  group_by(latent_var) %>% 
+  filter(corVal > 0.65) %>% #only positively correlated genes >0.65 or so...this is arbitrary
```

Comment:
I would remove this filter. GSEA is designed to take an ordered list. it will check for enrichment at the beginning or at the end, so ranking them all by correlation will take care of this.

Reply:
I agree, but doing this on the full list x full list by LV (drug targets to ranked viper proteins) yielded no significant results (even unadjusted p-value).

Reply:
Then I'd probably conclude that there is no enrichment...